src/wmtk/operations/tri_mesh/EdgeSplitWithTag.cpp

```diff
@@ -0,0 +1,98 @@
+#include "EdgeSplitWithTag.hpp"
+#include <wmtk/invariants/find_invariant_in_collection_by_type.hpp>
+
+#include <wmtk/TriMesh.hpp>
+#include <wmtk/invariants/TodoInvariant.hpp>
+#include "EdgeSplit.hpp"
+
+namespace wmtk::operations {
+
+
+void OperationSettings<tri_mesh::EdgeSplitWithTag>::initialize_invariants(const TriMesh& m)
+{
+    split_settings.initialize_invariants(m);
+    split_settings.invariants.add(std::make_shared<TodoInvariant>(m, split_todo));
+}
+
+bool OperationSettings<tri_mesh::EdgeSplitWithTag>::are_invariants_initialized() const
+{
+    return split_settings.are_invariants_initialized() &&
+           find_invariants_in_collection_by_type<TodoInvariant>(split_settings.invariants);
+}
+namespace tri_mesh {
+EdgeSplitWithTag::EdgeSplitWithTag(
+    Mesh& m,
+    const Tuple& t,
+    const OperationSettings<EdgeSplitWithTag>& settings)
+    : TriMeshOperation(m)
+    , TupleOperation(settings.split_settings.invariants, t)
+    , m_pos_accessor{m.create_accessor(settings.position)}
+    , m_vertex_tag_accessor{m.create_accessor(settings.vertex_tag)}
+    , m_edge_tag_accessor{m.create_accessor(settings.edge_tag)}
+    , m_split_todo_accessor{m.create_accessor(settings.split_todo)}
+    , m_settings{settings}
+{}
+std::string EdgeSplitWithTag::name() const
+{
+    return "edge_split_with_tag";
+}
+Tuple EdgeSplitWithTag::return_tuple() const
+{
+    return m_output_tuple;
+}
+bool EdgeSplitWithTag::before() const
```

Comment:
You don't need to override `before()`. It is already defined in the base class. Just remove `before` completely from your class.